### PR TITLE
Path Length Map example

### PR DESCRIPTION
**Pull request by kesshijordan**
Added documentation for #1114. I plan to put a preprint up to reference soon (hopefully this week), so I currently have a placeholder in the documentation.

### CODE REVIEW

**Comment by arokem** [comment]
Here's a good reference for why `jet` is not a great color-map:
https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/

File: doc/examples/path_length_map.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+from dipy.tracking import utils
+from dipy.tracking.local import LocalTracking
+from dipy.tracking.streamline import Streamlines
+from dipy.viz import actor, window
+from dipy.viz.colormap import line_colors
+from dipy.tracking.utils import path_length
+import nibabel as nib
+import numpy as np
+import os
+import nilearn.plotting as nip
```
[the remark this answers is not in the document]
Comment (kesshijordan) [reply doc/examples/path_length_map.py]: Sure. I changed it to matplotlib, modeled after the one in the LiFE example.

File: doc/examples/path_length_map.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+# save the WMPL as a nifti
+path_length_img = nib.Nifti1Image(wmpl.astype(np.float32), affine)
+nib.save(path_length_img, 'example_cc_path_length_map.nii.gz')
+
+# generate display of Path Length map using Nilearn
+t1_path = os.path.join(fetch_stanford_t1()[1], 't1.nii.gz')
+nip.plot_stat_map('example_cc_path_length_map.nii.gz',
+                  output_file='Path_Length_Map.png',
+                  bg_img=t1_path, symmetric_cbar=False,
+                  cmap='jet', threshold=1)
```
[the remark this answers is not in the document]
Comment (kesshijordan) [reply doc/examples/path_length_map.py]: I really like this explanation, too (see below). Absolutely... thanks for the catch : ) Viridis and the other perceptually uniform options are too dark at the bottom of the scale, which is important to stand out in this application, so I changed it to cool but am open to suggestions.

https://twitter.com/choldgraf/status/1006559050169872384?lang=en

**Comment by skoudoro** [comment]
I need to test it on windows (maybe tomorrow), but it will be good if someone else could look at this PR before merging it

File: doc/examples/path_length_map.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+# Visualize the streamlines and the Path Length Map base ROI
+# (in this case also the seed ROI)
+
+streamlines_actor = actor.line(streamlines, line_colors(streamlines))
+surface_opacity = 0.5
+surface_color = [0, 1, 1]
+seedroi_actor = actor.contour_from_roi(seed_mask, affine,
+                                       surface_color, surface_opacity)
+
+ren = window.ren()
```
Comment (skoudoro) [inline doc/examples/path_length_map.py]: here, it should be `window.Renderer()`.  Indeed, `window.ren() is deprecated`

File: doc/examples/path_length_map.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+ren.add(streamlines_actor)
+ren.add(seedroi_actor)
+
+"""
+If you set interactive to True (below), the rendering will pop up in an
+interactive window.
+"""
+
+interactive = False  # this works if it's True but black if False??
```
[the remark this answers is not in the document]
Comment (skoudoro) [reply doc/examples/path_length_map.py]: Strange... I never got an empty png under windows. Thanks for the notice, I will check Life tutorial

**Comment by kesshijordan** [comment]
Thanks for all your work on this, @skoudoro ! I addressed all your comments. I'll add the preprint when it goes up and that's the last bit, I think.

File: doc/examples/path_length_map.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+from dipy.viz.colormap import line_colors
+from dipy.tracking.utils import path_length
+import nibabel as nib
+import numpy as np
+import matplotlib as mpl
+from mpl_toolkits.axes_grid1 import AxesGrid
+
+"""
+First, we need to generate some streamlines and visualize. For a more complete
+description of these steps, please refer to the CSA Probabilistic Tracking and
```
[the remark this answers is not in the document]
Comment (kesshijordan) [reply doc/examples/path_length_map.py]: Thanks, @arokem ! I will do that

**Comment by arokem** [comment]
Superseded by #1649

**Pull request closed without merging**